test/test_jit.py

```diff
@@ -1259,6 +1259,30 @@ def fn(x, t):
         inputs = (torch.randn(1), (torch.randn(1), torch.randn(1)))
         self.checkTrace(fn, inputs)
 
+    def test_input_dict_flattens(self):
+        class Test(torch.nn.Module):
+            def forward(self, d):
+                return d['x'] + d['y']
+
+        inputs = {'x': torch.rand(3, 4), 'y': torch.rand(3, 4)}
+        module = torch.jit.trace(Test(), inputs)
+        kinds = [node.kind() for node in module.graph.nodes()]
```

Comment:
We have a FileCheck-like utility for asserting graph structure; you should use that instead of the python IR bindings.